### DISCUSSION TITLE
Retry with exponential back-off

### DISCUSSION THREAD

**Discussion by slavafomin** [sindresorhus/awesome-nodejs#1115] [Ideas]
Hey!

I guess it will be nice to add to the list some utility library that simplifies retrying and supports exponential back-off strategy. This is often used to make HTTP requests more stable, especially on the clients with flaky connection.

**Comment by slavafomin** [comment]
After looking into the options, it looks like [`sindresorhus/p-retry`](https://github.com/sindresorhus/p-retry) would be the best choice:

**Stars**: 373
**Last release**: 15 days ago
**Weekly Downloads**: 8 451 795
**Bundlephobia**: 4.5 KB minified
**TypeScript Support**: Yes

Uses a popular and mature implementation: [`tim-kos/node-retry`](https://github.com/tim-kos/node-retry) under the hood.

Looks like @sindresorhus did it again :)

**Comment by Divadody** [comment]
Good
